packages/mssql/src/index.ts

```diff
@@ -0,0 +1,403 @@
+/**
+ * @since 1.0.0
+ */
+import { Tag } from "@effect/data/Context"
+import * as Debug from "@effect/data/Debug"
+import type { Duration } from "@effect/data/Duration"
+import { minutes } from "@effect/data/Duration"
+import { pipe } from "@effect/data/Function"
+import * as Config from "@effect/io/Config"
+import * as ConfigSecret from "@effect/io/Config/Secret"
+import * as Effect from "@effect/io/Effect"
+import * as Layer from "@effect/io/Layer"
+import * as Pool from "@effect/io/Pool"
+import type { Scope } from "@effect/io/Scope"
+import * as Client from "@sqlfx/sql/Client"
+import type { Connection } from "@sqlfx/sql/Connection"
+import { SqlError } from "@sqlfx/sql/Error"
+import type { Custom, Primitive as _Primitive } from "@sqlfx/sql/Statement"
+import * as Statement from "@sqlfx/sql/Statement"
+import * as transform from "@sqlfx/sql/Transform"
+import * as Tedious from "tedious"
+
+export {
+  /**
+   * Column renaming helpers.
+   *
+   * @since 1.0.0
+   */
+  transform,
+}
+
+/**
+ * @category model
+ * @since 1.0.0
+ */
+export interface TediousClient extends Client.Client {
+  readonly config: TediusClientConfig
+}
+
+/**
+ * @category tag
+ * @since 1.0.0
+ */
+export const tag = Tag<TediousClient>()
+
+/**
+ * @category constructor
+ * @since 1.0.0
+ */
+export interface TediusClientConfig {
+
+  readonly domain?: string
+  readonly server?: string
+  readonly port?: number
+  readonly database?: string
+  readonly username?: string
+  readonly password?: ConfigSecret.ConfigSecret
+  readonly connectTimeout?: Duration
+
+  readonly minConnections?: number
+  readonly maxConnections?: number
+  readonly connectionTTL?: Duration
+
+  readonly transformResultNames?: (str: string) => string
+  readonly transformQueryNames?: (str: string) => string
+}
+
+const escape = Statement.defaultEscape('')
+
+/**
+ * @category constructor
+ * @since 1.0.0
+ */
+export const make = (
+  options: TediusClientConfig,
+): Effect.Effect<Scope, never, TediousClient> =>
+  Effect.gen(function*(_) {
+    const compiler = makeCompiler(options.transformQueryNames)
+
+    const transformRows = Client.defaultRowTransform(
+      options.transformResultNames!,
+    )
+
+    const makeConnection = pipe(
+      Effect.acquireRelease(
+        Effect.tap(Effect.sync(() =>
+          new Tedious.Connection({
```

Comment:
They have may more options, maybe interesting, maybe not.

Reply:
I feel like there are a bunch of missing config items, but we can cross that bridge later.